### ISSUE TITLE
'n' LBLD objects for 'n' coordinate pair ?

### ISSUE THREAD

**Issue by sagarpreet-chadha** [publiclab/leaflet-blurred-location-display#66]
From #64 ,

>  does options.locations auto-generate the BlurredLocations? Why do we make only one blurredLocation? It kind of seems like we should make one for each coordinate pair, using this syntax?

Hi @jywarren ,

- So currently we have this `options.locations` property which takes an array of coordinates as input and then performs the normal LBLD functions on them .

- We are making only one LBL and one LBLD object currently . We pass the `map` object as options to the `LBL object` , and then this `LBL object` in `options_display` to `LBLD object` .

I think `options.locations` property gives user more flexibility . So if one do not want to pass their `API URL` and `JSON parser` into the LBLD library (_as we do not support XML parsing and others_) , so instead one can pass the coordinates array (locations array) directly into the LBLD object . Makes sense ?

Kindly elaborate more on this issue :smile: !  Thanks !

**Comment by sagarpreet-chadha** [comment]
Hi @jywarren , 
I have changed the LBL declaration syntax in PR : https://github.com/publiclab/leaflet-blurred-location/pull/195 .
 The current declaration of LBLD is pretty simple 😄  : 

`var blurredLocationDisplay = new BlurredLocationDisplay(options_display);`

Should we do any change to LBLD as well ? Kindly elaborate this issue . Thanks !

**Comment by jywarren** [comment]
Yeah i think it's fine - however does it set up a Leaflet map, or how do we
connect it to a leaflet map? Especially, say, if we have more than one on a
given page?

**Comment by sagarpreet-chadha** [comment]
So after https://github.com/publiclab/leaflet-blurred-location/pull/195 gets merged ,
It would be a 3 step process to add LBLD to maps : 

1.) Make **standard leaflet object**  : 
`var map123 = L.map('map').setView([23,77] , 3) ;`

2.) Pass this leaflet object to **LBL** : 
`var blurredLocation = new BlurredLocation(options) ;`
`blurredLocation.addTo(map123) ;`

3.) Pass **LBL object** in `options_display` to **LBLD object** (no need to pass leaflet map object) : 
`var blurredLocationDisplay = new BlurredLocationDisplay(options_display);`

- So for multiple maps , we will have multiple standard map objects and corresponding LBL and LBLD objects . 

I will write a detailed documentation explaining this in `Readme`  :)
